.travis.yml

```diff
@@ -28,17 +28,17 @@ addons:
       - llvm-toolchain-precise
 
 env:
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Ninja"
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
-
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=gcc CMAKE_GENERATOR="Ninja"
-  - COMPILER=gcc CMAKE_GENERATOR="Unix Makefiles"
+  # Test all generators and Clang + GCC
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=gcc         CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=gcc         CMAKE_GENERATOR="Unix Makefiles"
+
+  # Benchmarks
+  - BENCHMARK=true COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=gcc         CMAKE_GENERATOR="Unix Makefiles"
```

Comment:
Sorry, this thread somehow went under my radar.


> More importantly, if several jobs are pushing data to gh-pages, which job will build the front page that shows everything?

No worries, Hana currently does just this to update its own benchmarks and upload them to gh-pages. Basically, each job will clone the repository and checkout the gh-pages branch. Then, each job

1. Generates its part of the benchmarks
2. Updates gh-pages (perhaps another job pushed something while this job was generating the benchmarks).
3. Commit the new changes on gh-pages
4. Push

Steps 2-4 are very quick to perform, so there is only a very small possibility that two jobs get in conflict. And even if that happens, we actually try the 2-4 steps three times before giving up and failing. So this can fail, but we'd have to be very unlucky. It has not failed yet for Hana, anyway.

> It definitely is, but it also adds a lot of complexity.

I don't think it adds too much complexity. Basically,

1. We must support all the relevant compilers in the benchmarks.
2. We must be able to pick which compiler we're displaying on the gh-pages site.

(1) is done since this PR is merged, and I don't think (2) should be complicated. But I'm cautious of not taking too large bites, and if this gets out of hand, rest assured that I'm prepared to drop this feature so we can progress, at least temporarily.